samples/bluetooth/fast_pair/locator_tag/src/main.c

```diff
@@ -343,31 +370,29 @@ static void fmdn_provisioning_state_changed(bool provisioned)
 			FACTORY_RESET_TRIGGER_KEY_STATE_MISMATCH,
 			K_SECONDS(FACTORY_RESET_DELAY));
 
+		first_state_change_after_bootup = false;
```

Comment:
This is not the best way to do this as the user needs to keep in mind to add this line in case of another early return.

Reply:
Avoiding early returns and using all of the information explicitly to assign `fp_adv` requests at the end could make this code easier to understand.